### PR TITLE
[CORE] Fix incorrect precision of decimal literal

### PR DESCRIPTION
**Pull request by jiangjiangtian**
For SQL as follows:
```
select (col0 / (col1 + 0.00000001)) from table;
```
In this case, col0 and col1 are 0.
The result may be NULL. The reason is that the result of `Decimal(0.00000001).toString()` is "1E-8", which will make the new precision be 4.
Therefore, we use `toPlainString` here to prevent scientific notation.

### CODE REVIEW

**Comment by jiangjiangtian** [comment]
@kecookier

**Comment by jiangjiangtian** [comment]
@rui-mo Can you review this PR?
I hava a question: why do we have this rescale in gluten? I can't find the same logic in Spark. I have read the comment and I still can't have a good understanding of it.
Thanks!

**Comment by rui-mo** [comment]
Hi @jiangjiangtian, this adjustment, as I recall, is for the situation where one performs an arithmetic operation between a decimal and a number. In this instance, the precision and scale of the decimal input acquired from Spark are (38, 18), which are inconsistent with the real values. In order to produce accurate results, we need additional logic to extract the accurate precision and scale that native computing requires.

**Review by rui-mo — COMMENTED** [review]
Thanks for the fix. Would you like to add the buggy case as a unit test? Thanks.

**Comment by jiangjiangtian** [comment]
> Hi @jiangjiangtian, this adjustment, as I recall, is for the situation where one performs an arithmetic operation between a decimal and a number. In this instance, the number is converted to decimal and its precision and scale acquired from Spark are (38, 18), which are inconsistent with the real values. E.g., in the case you mentioned, `Decimal(0.00000001)` should have a precision and scale of (8, 8) instead of (38, 18). In order to produce accurate results, we need additional logic to extract the accurate precision and scale that native computing requires.
> 
> Perhaps you could help confirm if it is the case for the example you provided. Thanks.

@rui-mo Thanks! It seems that Spark doesn't have the logic. I don't know why Spark doesn't need this logic.

In my case, the type of the literal `0.00000001` is `Decimal(19, 8)`. After the adjustment, the type is still `Decimal(19, 8)`. Because the string representation of the decimal contains `.` and the decimal is not a valid long number. Perhaps we should not return the original precision and scale in the end of the function. https://github.com/apache/incubator-gluten/blob/main/gluten-core/src/main/scala/org/apache/gluten/utils/DecimalArithmeticUtil.scala#L110

**Comment by jiangjiangtian** [comment]
> Thanks for the fix. Would you like to add the buggy case as a unit test?

I add a unit test. Is there anything that I need to add or edit? Thanks.

**Comment by rui-mo** [comment]
> Because the string representation of the decimal contains . and the decimal is not a valid long number. 

Thanks for reminding me of this. I just remembered that this adjustment is typically for the arithmetic operation between a decimal and an integer/bigint. In your case, the literal is double, so to return the original precision and scale should be fine.

File: backends-velox/src/test/scala/org/apache/gluten/execution/MiscOperatorSuite.scala [hunk 1]
```diff
@@ -2086,4 +2086,12 @@ class MiscOperatorSuite extends VeloxWholeStageTransformerSuite with AdaptiveSpa
     val df2 = runQueryAndCompare("SELECT round(cast(0.19324999999999998 as double), 2)") { _ => }
     checkLengthAndPlan(df2, 1)
   }
+
+  test("Fix wrong rescale") {
+    withTable("test") {
+      sql("create table test (col0 decimal(10, 0), col1 decimal(10, 0)) using parquet")
+      sql("insert into test values (0, 0)")
+      runQueryAndCompare("select col0 / (col1 + 1E-8) from test") { _ => }
```
[the remark this answers is not in the document]
Comment (jiangjiangtian) [reply backends-velox/src/test/scala/org/apache/gluten/execution/MiscOperatorSuite.scala]: Fixed.

**Review by rui-mo — APPROVED** [review]

**Pull request merged**